braindecode/datautil/windowers.py

```diff
@@ -65,17 +65,28 @@ def create_windows_from_events(
         trial_start_offset_samples, trial_stop_offset_samples,
         supercrop_size_samples, supercrop_stride_samples)
 
+    infer_mapping = True if mapping is None else False
+
     list_of_windows_ds = []
     for ds in concat_ds.datasets:
-        if mapping is None:
+        if infer_mapping:
             # mapping event descriptions to integers from 0 on
-            mapping = {v: k for k, v in enumerate(
-                np.unique(ds.raw.annotations.description))}
-
-        events, _ = mne.events_from_annotations(ds.raw, mapping)
+            unique_events = np.unique(ds.raw.annotations.description)
+            if mapping is None:
+                mapping = {v: k for k, v in enumerate(unique_events)}
+            else:
+                max_id_mapping = max(list(mapping.values())) + 1
+                unique_events = [x for x in unique_events if x not in mapping]
+                mapping.update(
+                    {v: k + max_id_mapping for k, v in enumerate(unique_events)}
+                )
+
+        events, events_id = mne.events_from_annotations(ds.raw, mapping)
```

Comment:
I would refactor overall to:
```python
# If user did not specify mapping,
# we extract all events from all datasets
# and map them to increasing integers
# starting form 0
infer_mapping = mapping is None
if infer_mapping:
	mapping = {}

list_of_windows_ds = []
for ds in concat_ds.datasets:
    if infer_mapping:
	unique_events = np.unique(ds.raw.annotations.description)
        new_unique_events = [x for x in unique_events if x not in mapping]
        # mapping event descriptions to integers from 0 on
	if len(mapping) == 0:
		max_id_mapping = 0
	else:
	        max_id_mapping = max(list(mapping.values())) + 1
        mapping.update(
            {v: k + max_id_mapping for k, v in enumerate(new_unique_events)}
```
Found code a bit hard to read quickly before, like did not understand mapping will change for example